packages/grid/src/EditableGridModel.ts

```diff
@@ -0,0 +1,96 @@
+import GridRange from './GridRange';
+import { ModelIndex } from './GridMetrics';
+import GridModel from './GridModel';
+
+export function isEditableGridModel(
+  model: GridModel
+): model is EditableGridModel {
+  return (model as EditableGridModel)?.isEditable !== undefined;
+}
+
+/**
+ * Edit operation when applying multiple edits
+ */
+export type EditOperation = {
+  /** Column to set the value for */
+  column: ModelIndex;
+
+  /** Row to set the value for */
+  row: ModelIndex;
+
+  /** Text value to set */
+  text: string;
+
+  /** @deprecated use `column` instead */
+  x?: ModelIndex;
+  /** @deprecated use `row` instead */
+  y?: ModelIndex;
+};
+
+/**
+ * Model for an editable grid
+ */
+export interface EditableGridModel extends GridModel {
+  isEditable: boolean;
+
+  /**
+   * Check if a given range is editable
+   * @param range The range to check if it is editable
+   * @returns True if the range is editable
+   */
+  isEditableRange(range: GridRange): boolean;
+
+  /**
+   * Get the edit text for a cell as a string
+   * @param column Column to get
+   * @param row Row to get
+   * @returns The value to use for editing
+   */
+  editValueForCell(column: ModelIndex, row: ModelIndex): string;
+
+  /**
+   * Set value in an editable table
+   * @param column Column to set
+   * @param row Row to set
+   * @param value The value to set
+   * @returns A promise that resolves successfully when the operation is complete, or rejects if there's an error
+   */
+  setValueForCell(
+    column: ModelIndex,
+    row: ModelIndex,
+    value: string
+  ): Promise<void>;
+
+  /**
+   * Set value in an editable table
+   * @param ranges The ranges to set
+   * @param value The value to set
+   * @returns A promise that resolves successfully when the operation is complete, or rejects if there's an error
+   */
+  setValueForRanges(ranges: GridRange[], value: string): Promise<void>;
+
+  /**
+   * Apply edits to the model
+   * @param edits Edits to apply to the model
+   * @returns A promise that resolves successfully when the operation is complete or rejects if there's an error
+   */
+  setValues(edits: EditOperation[]): Promise<void>;
+
+  /**
+   * Check if a text value is a valid edit for a cell
+   * @param column Column to check
+   * @param row Row to check
+   * @param value Value to check if it's a valid value or not
+   * @returns True if it's a valid value, false otherwise
+   */
+  isValidForCell(column: ModelIndex, row: ModelIndex, value: string): boolean;
+
+  /**
+   * Delete ranges from an editable grid
+   * @param ranges The ranges to delete
+   * @returns A promise that resolves successfully when the operation is complete or rejects if there's an error
+   */
+  delete(ranges: GridRange[]): Promise<void>;
```

Comment:
Yes

Reply:
Can you clarify which behavior in the comment?

I'm assuming it nulls out the values in the selection and not the Excel "delete range" which then asks if you want to shift data left/up/etc.

Reply:
True. And in fact, I don't even have this wired up in @deephaven/grid at all, so I'm just going to move the delete function into `IrisGridModel`... until we have some way from Grid itself to do this delete row function, it doesn't make sense to have in grid I don't think. And then it can take a param to ask if it should shift.
And I'm not going to change the signature or typescriptify IrisGridModel in this change either - will decide to clean that up at this time.

To null/blank out the values in a selection we already do that by calling `setValueForRanges`.

Anyway, good catch!